### PR TITLE
test: fuzz: migrate oss-fuzz fuzzer.

### PR DESCRIPTION
**Pull request by DavidKorczynski**
Cross-referencing this PR on oss-fuzz https://github.com/google/oss-fuzz/pull/5687 it would be great to move the fuzzer upstream. Once the fuzzer is upstream I will fixup the infra on the oss-fuzz side.

### CODE REVIEW

File: test/fuzz/fuzz_parser.c [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <stdbool.h>
+
+int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size) {
+	char *new_str = (char *)malloc(size+1);
+	if (new_str == NULL){
+			return 0;
```
Comment (lfittl) [inline test/fuzz/fuzz_parser.c]: Minor nitpick, but this line seems to be indented 1 tab too much :)

**Review by lfittl — COMMENTED** [review]
@DavidKorczynski Perfect, thanks!

Could you also confirm that this contribution is made by you personally and subject to the pg_query license? (https://github.com/pganalyze/libpg_query/blob/13-latest/LICENSE)

(since the same code was also [added upstream](https://github.com/google/oss-fuzz/pull/5687) as Apache 2.0 licensed, I want to clarify that for the historic record)

File: test/fuzz/fuzz_parser.c (hunk 1 above)
Comment (DavidKorczynski) [reply test/fuzz/fuzz_parser.c]: Ah thanks, should be fixed now!

**Comment by DavidKorczynski** [comment]
I confirm that all code in this PR is written by me personally and perfectly fine that it will be subject to pg_query license

**Review by lfittl — APPROVED** [review]

**Comment by lfittl** [comment]
@DavidKorczynski Perfect, thanks for confirming. Just running the tests now and will merge shortly :)

**Comment by DavidKorczynski** [comment]
Thanks @lfittl !

**Pull request merged**